### PR TITLE
add error checks for datadog exporter

### PR DESCRIPTION
**Pull request by DiptoChakrabarty**
**Description:** <Describe what has changed.>

adds error check in datadogexporters

**Link to tracking Issue:** #9749 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/datadogexporter/factory_test.go [hunk 1]
```diff
@@ -580,7 +579,10 @@ func TestOnlyMetadata(t *testing.T) {
 
 	err = expTraces.Start(ctx, nil)
 	assert.NoError(t, err)
-	defer expTraces.Shutdown(ctx)
+	defer func() {
+		err = expTraces.Shutdown(ctx)
+		assert.NoError(t, err)
```
Comment (bogdandrutu) [inline exporter/datadogexporter/factory_test.go:584]:
```suggestion
		assert.NoError(t, expTraces.Shutdown(ctx))
```

File: exporter/datadogexporter/internal/metrics/consumer_test.go [hunk 2]
```diff
@@ -71,7 +70,8 @@ func TestRunningMetrics(t *testing.T) {
 
 	ctx := context.Background()
 	consumer := NewConsumer()
-	tr.MapMetrics(ctx, ms, consumer)
+	err := tr.MapMetrics(ctx, ms, consumer)
+	assert.NoError(t, err)
```
Comment (bogdandrutu) [inline exporter/datadogexporter/internal/metrics/consumer_test.go:74]:
```suggestion
	assert.NoError(t, tr.MapMetrics(ctx, ms, consumer))
```

File: exporter/datadogexporter/internal/metrics/consumer_test.go [hunk 3]
```diff
@@ -115,7 +115,8 @@ func TestTagsMetrics(t *testing.T) {
...
```
Comment (bogdandrutu) [inline exporter/datadogexporter/internal/metrics/consumer_test.go:119]:
```suggestion
	assert.NoError(t, tr.MapMetrics(ctx, ms, consumer))
```

File: exporter/datadogexporter/traces_exporter_test.go [hunk 4]
```diff
@@ -91,7 +90,10 @@ func testTracesExporterHelper(td ptrace.Traces, t *testing.T) []string {
 
 	assert.NoError(t, err)
 
-	defer exporter.Shutdown(context.Background())
+	defer func() {
+		err = exporter.Shutdown(context.Background())
+		assert.NoError(t, err)
```
Comment (bogdandrutu) [inline exporter/datadogexporter/traces_exporter_test.go:95]:
```suggestion
		assert.NoError(t, exporter.Shutdown(context.Background()))
```

File: exporter/datadogexporter/internal/model/translator/sketches_test.go [hunk 5]
```diff
@@ -134,7 +133,8 @@ func TestHistogramSketches(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			md := fromCDF(test.cdf)
 			consumer := &sketchConsumer{}
-			tr.MapMetrics(ctx, md, consumer)
+			err := tr.MapMetrics(ctx, md, consumer)
+			assert.NoError(t, err)
```
Comment (bogdandrutu) [inline exporter/datadogexporter/internal/model/translator/sketches_test.go:137]:
```suggestion
			assert.NoError(t, tr.MapMetrics(ctx, md, consumer))
```

File: exporter/datadogexporter/internal/model/translator/sketches_test.go [hunk 6]
```diff
@@ -305,7 +305,8 @@ func TestExactSumCount(t *testing.T) {
 		t.Run(testInstance.name, func(t *testing.T) {
 			md := testInstance.getHist()
 			consumer := &sketchConsumer{}
-			tr.MapMetrics(ctx, md, consumer)
+			err := tr.MapMetrics(ctx, md, consumer)
+			assert.NoError(t, err)
```
Comment (bogdandrutu) [inline exporter/datadogexporter/internal/model/translator/sketches_test.go:309]:
```suggestion
			assert.NoError(t, tr.MapMetrics(ctx, md, consumer))
```

File: exporter/datadogexporter/internal/model/translator/sketches_test.go [hunk 7]
```diff
@@ -363,7 +364,8 @@ func TestInfiniteBounds(t *testing.T) {
...
```
Comment (bogdandrutu) [inline exporter/datadogexporter/internal/model/translator/sketches_test.go:368]:
```suggestion
			assert.NoError(t, tr.MapMetrics(ctx, md, consumer))
```

File: exporter/datadogexporter/factory_test.go (hunk 1 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/factory_test.go]: done

File: exporter/datadogexporter/internal/metrics/consumer_test.go (hunk 2 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/internal/metrics/consumer_test.go]: done

File: exporter/datadogexporter/internal/metrics/consumer_test.go (hunk 3 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/internal/metrics/consumer_test.go]: done

File: exporter/datadogexporter/traces_exporter_test.go (hunk 4 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/traces_exporter_test.go]: done

File: exporter/datadogexporter/internal/model/translator/sketches_test.go (hunk 5 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/internal/model/translator/sketches_test.go]: done

File: exporter/datadogexporter/internal/model/translator/sketches_test.go (hunk 6 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/internal/model/translator/sketches_test.go]: done

File: exporter/datadogexporter/internal/model/translator/sketches_test.go (hunk 7 above)
Comment (DiptoChakrabarty) [reply exporter/datadogexporter/internal/model/translator/sketches_test.go]: done

**Review by mx-psi — APPROVED** [review]

**Pull request merged**